### PR TITLE
Fix autocorrect with parenthesis on same line

### PR DESCRIPTION
**Pull request by nikolai-b**
Basic fix to add a new line.  I was not sure how to get the alignment correct as I wanted to run Style/MultilineMethodCallIndentation autocorrect afterwards but I wasn't sure of the best way to do this.

### CODE REVIEW

**Comment by bbatsov** [comment]
@jonas054 Can you take a look here as well?

This certain needs a changelog entry and a green build.

File: lib/rubocop/cop/style/multiline_memoization.rb [hunk 1]
```diff
@@ -52,13 +52,28 @@ def on_or_asgn(node)
         private
 
         def autocorrect(node)
+          b = node.loc.begin
```
Comment (bbatsov) [inline lib/rubocop/cop/style/multiline_memoization.rb]: I'd use more descriptive var names.

File: lib/rubocop/cop/style/multiline_memoization.rb [hunk 2]
```diff
@@ -52,13 +52,28 @@ def on_or_asgn(node)
...
-              corrector.replace(node.loc.begin, 'begin')
-              corrector.replace(node.loc.end, 'end')
+              replacement = if newline_after?(node, b.end_pos)
+                              'begin'
+                            else
+                              "begin\n" + (' ' * b.begin_pos)
+                            end
+              corrector.replace(b, replacement)
+
+              replacement = if newline_before?(node)
```
Comment (bbatsov) [inline lib/rubocop/cop/style/multiline_memoization.rb]: I don't like reusing variable names - makes it a bit confusing to read the code.

File: lib/rubocop/cop/style/multiline_memoization.rb [hunk 3]
```diff
@@ -71,6 +86,15 @@ def bad_rhs?(rhs)
             rhs.kwbegin_type?
           end
         end
+
+        def newline_after?(node, pos)
+          node.source_range.source_buffer.source[pos] == "\n"
```
Comment (bbatsov) [inline lib/rubocop/cop/style/multiline_memoization.rb]: @jonas054 any idea how to simplify those methods?

Comment (jonas054) [inline lib/rubocop/cop/style/multiline_memoization.rb]: Actually I think the code needs to be more complicated. :smile: @nikolai-b Please add comments after the opening `(` in the "code with offense" examples and make sure the comment ends up in the corresponding position in the auto-corrected code.

Comment (nikolai-b) [inline lib/rubocop/cop/style/multiline_memoization.rb]: @jonas054 to make it line up do is there another autocorrect I can call or do I have to do it implement lining up here?

Comment (jonas054) [inline lib/rubocop/cop/style/multiline_memoization.rb]: We never call `autocorrect` methods in other cops, as far as I know. I think that would be messy.

If it's not too difficult, do the lining up here. If it's overly complicated, just leave it and another cop will step in during the full run.

I don't remember if @bbatsov has stated a clear preference, if it's okay for autocorrect to leave the code a bit untidy and rely on other cops.

**Comment by bbatsov** [comment]
Closing due to lack of activity.

**Pull request closed without merging**